### ISSUE TITLE
is there a way to tune categorical hyperparameters?

### ISSUE THREAD

**Issue by gxr1894** [LGE-ARC-AdvancedAI/auptimizer#14]
I noticed that it is easy to 'auptimize' over hyper-parameters which are integer or float types, but is it possible to optimize over categorical hyper-parameters, for example loss type, weight decay type, etc?

**Comment by guojunyao419** [comment]
Great observation! 

Yes, you can also optimize categorical hyper-parameters, if they can be represented by strings. You can use `sequence` or `random` proposer, and list all the values in string in `range`, and then set the `type` to be `choice`. 

Example:
`"parameter_config": [
   {
     "name": "loss",
     "range": [
       "mse",
       "cross-entropy"
     ],
     "type": "choice"
   }
 ]`
